### ISSUE TITLE
`init-declarations` false positive at `for-in/of`.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#3202]
## test.js

```js
/* eslint init-declarations: 2 */
var a = 0;

for (var b in []) {}
```

## Command & Actual

```
> eslint -v
v1.0.0-rc-3
> eslint test.js --no-eslintrc

test.js
  4:10  error  Variable 'b' should be initialized on declaration  init-declarations

✖ 1 problem (1 error, 0 warnings)
```

## Expected

No errors.
